jupyterhub/apihandlers/users.py

```diff
@@ -193,6 +197,52 @@ def delete(self, name):
         status = 202 if user.stop_pending else 204
         self.set_status(status)
 
+
+class UserCreateMultiServerAPIHandler(APIHandler):
+    """Create multi single-user server
+    
+    This handler should be used when c.JupyterHub.allow_multiple_servers = True
+    """
+    @gen.coroutine
+    @admin_or_self
+    def post(self, name):
+        user = self.find_user(name)
+        if user.running:
+            # include notify, so that a server that died is noticed immediately
+            state = yield user.spawner.poll_and_notify()
+            if state is None:
+                raise web.HTTPError(400, "%s's server is already running" % name)
+
+        options = self.get_json_body()
+        yield self.spawn_single_user(user, options=options)
+        status = 202 if user.spawn_pending else 201
+        self.set_status(status)
+
+
+class UserDeleteMultiServerAPIHandler(APIHandler):
```

Comment:
what about `UserCreateMultiServerAPIHandler` ?
do you prefer to keep the this name or `NamedServer` ?